lib/new_relic/agent.rb

```diff
@@ -649,6 +649,43 @@ def add_custom_span_attributes(params)
       end
     end
 
+    # Add custom attributes to log events for the current agent instance.
+    #
+    # @param [Hash] params    A Hash of attributes to attach to log
+    #                         events. The agent accepts up to 240 custom
+    #                         log event attributes.
+    #
+    #                         Keys will be coerced into Strings and must
+    #                         be less than 256 characters. Keys longer
+    #                         than 255 characters will be truncated.
+    #
+    #                         Values may be Strings, Symbols, numeric
+    #                         values or Booleans and must be less than
+    #                         4095 characters. If the value is a String
+    #                         or a Symbol, values longer than 4094
+    #                         characters will be truncated. If the value
+    #                         exceeds 4094 characters and is of a
+    #                         different class, the attribute pair will
+    #                         be dropped.
+    #
+    #                         This API can be called multiple times.
+    #                         If the same key is passed more than once,
+    #                         the value associated with the last call
+    #                         will be preserved.
+    #
+    #                         Attribute pairs with empty or nil contents
+    #                         will be dropped.
+    # @api public
+    def add_custom_log_attributes(params)
+      record_api_supportability_metric(:add_custom_log_attributes)
+
+      if params.is_a?(Hash)
```

Comment:
I'm not familiar with our API coding best practices, so maybe this is covered somewhere... but as a customer I'm thinking it would be really cool if you returned one thing to me if my call was successful (perhaps return my original input? or perhaps the current full list of attributes?) and another thing if the call failed (perhaps return `nil`).

Reply:
> Would you like me to change the return value to match your suggestion?

Given that a) we don't have a standard, b) you've modeled this new API closely off of a fairly similar one, and c) our communication back to the caller is traditionally done via logged messages, no. Thank you for the explanation!